11-lorawan/11-lorawan.md

```diff
@@ -0,0 +1,188 @@
+
+### Testing procedure
+
+Build and flash the `examples/lorawan` application with the following command
+
+    $ make BOARD=<board> DEVEUI=<device eui> APPEUI=<application eui> APPKEY=<application key> -C examples/lorawan flash term
+
+_Note:_ The application can be tested on IoT-LAB with a device register on
+TheThingsNetwork backend: start an experiment with one of the st-lrwan1 boards
+and pass `IOTLAB_NODE=auto-ssh` to the make command.
+
+### Result
+
+With valid device EUI, application EUI and application key, the example
+application can join the backend. Messages are received on the backend every
+~20s.
+
+
+Task #02 - OTAA join procedure
+==============================
+
+### Description
+
+OTAA join procedure and uplink with datarates indexes 0, 3, 5 (others are optional)
+
+### Testing procedure
+
+- Build and flash the `tests/pkg_semtech-loramac`
+
+      $ make BOARD=<board> -C tests/pkg_semtech-loramac flash term
+
+- Configure the devive EUI, application EUI, application key
+
+      > loramac set deveui 0011223344556677
+      > loramac set appeui 0011223344556677
+      > loramac set devkey 00112233445566770011223344556677
+
+- Configure the desired datarate index
+
+      > loramac set dr <datarate index>
+
+- Join the network using OTAA
+
+      > loramac join otaa
+
+- Send confirmed messages on port 123
+
+      > loramac tx <message> cnf 123
+
+- Send unconfirmed messages on port 42
+
+      > loramac tx <message> uncnf 42
+
+- Reboot the node after each datarate test
+
+### Result
+
+All joins are successful, messages are sent with success and correctly received
+on the LoRaWAN backend (same datarate, same port).
+TX attempts within the dutycycle are skipped (very likely to happen with low
+datarates). Messages sent from the backend to the device are received.
+The device doesn't hang: the shell is responsive after the send/receive is
+finished.
+
+
+Task #03 - ABP join procedure
+=============================
+
+ABP join procedure and uplinks with datarates indexes 0, 3, 5 (others are optional)
+
+### Testing procedure
+
+- Build and flash the `tests/pkg_semtech-loramac`
+
+      $ make BOARD=<board> -C tests/pkg_semtech-loramac flash term
+
+- Configure the devive EUI, application EUI, application key
+
+      > loramac set deveui 0011223344556677
+      > loramac set appeui 0011223344556677
+      > loramac set devaddr 00112233
+      > loramac set appskey 00112233445566770011223344556677
+      > loramac set nwkskey 00112233445566770011223344556677
+
+- Configure the desired datarate index
+
+      > loramac set dr <datarate index>
+
+- If using TheThingsNetwork backend, configure the RX2 window datarate
+  parameter:
+
+      > loramac set rx2_dr 3
+
+- Join the network using ABP
+
+      > loramac join abp
+
+- Send confirmed messages on port 123
+
+      > loramac tx <message> cnf 123
+
+- Send unconfirmed messages on port 42
+
+      > loramac tx <message> uncnf 42
+
+- Reboot the node after each datarate test
```

Comment:
I would add here that's required to reset the frame counters in TTN. Otherwise the backend will ignore all uplink packets after reset

Reply:
I wrote this with STM32 L0/L1 in mind: they provide an EEPROM so the uplink counter is stored automatically and there's no need to reset the frame counter in TTN. I'll add a note about that.